documentation/development-docs/patterns-for-standard-cmdlets.md

```diff
@@ -0,0 +1,996 @@
+        [Parameter(Mandatory = false, ParameterSetName = "UpdateByInputObjectParameterSet")]
+        public Type1 Property1 { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = "UpdateByFooNameParameterSet")]
+        [Parameter(Mandatory = true, ParameterSetName = "UpdateByResourceIdParameterSet")]
+        [Parameter(Mandatory = false, ParameterSetName = "UpdateByInputObjectParameterSet")]
+        public Type2 Property2 { get; set; }
+
+        // Excluding other property parameters
+
+        public override void ExecuteCmdlet()
+        {
+            if (this.IsParameterBound(c => c.InputObject))
+            {
+                this.ResourceGroupName = this.InputObject.ResourceGroupName;
+                this.Name = this.InputObject.Name;
+                this.Property1 = this.IsParameterBound(c => c.Property1) ? this.Property1 : this.InputObject.Property1;
+                this.Property2 = this.IsParameterBound(c => c.Property2) ? this.Property2 : this.InputObject.Property2;
+                ...
+            }
+
+            if (this.IsParameterBound(c => c.ResourceId))
+            {
+                var resourceIdentifier = new ResourceIdentifier(this.ResourceId);
+                this.ResourceGroupName = resourceIdentifier.ResourceGroupName;
+                this.Name = resourceIdentifier.ResourceName;
+            }
+
+            Foo foo = null;
+            try
+            {
+                foo = this.MySDKClient.Foo.Get(this.ResourceGroupName, this.Name);
+            }
+            catch
+            {
+                foo = null;
+            }
+
+            if (foo == null)
+            {
+                throw new Exception(string.Format("A Foo with name '{0}' in resource group '{1}' does not exist. Please use New-AzureRmFoo to create a Foo with these properties.", this.Name, this.ResourceGroupName));
+            }
+
+            foo.Property1 = this.Property1;
+            foo.Property2 = this.Property2;
+            ...
+
+            if (this.ShouldProcess(this.Name, string.Format("Updating Foo '{0}' in resource group '{1}'.", this.Name, this.ResourceGroupName)))
+            {
+                var result = new PSFoo(this.MySDKClient.Foo.Update(this.ResourceGroupName, this.Name, foo)
+                WriteObject(result);
+            }
+        }
+    }
+}
+```
+
+</p>
+</details>
+
+### `Update-*` cmdlet
+
+All top-level resources should have an `Update-*` cmdlet that allows users to update an existing resource _if the API follows `PATCH` semantics_. If the API supports `PUT` semantics, then the cmdlet should be `Set-*` (see above). The user can update an existing resource by providing all identity properties, the resource id, or the object representation of the resource. This cmdlet should implement `SupportsShouldProcess` to allow users to provide the `-WhatIf` parameter and see what the result of executing the cmdlet is without the resource actually being updated.
```

Comment:
Note that cmdlets shoule not implement an Update cmdlet if the only property that can be updates through the resource's PATCH API is Tags

Reply:
Added a note about this.